custbaud_osx.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * custbaud_osx.c
+ *
+ * Custom baud rate support for OSX.
+ *
+ * by Joe Merten (https://github.com/Joe-Merten www.jme.de)
+ *
+ * ATTENTION: OSX specific stuff!
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+ * USA
+ */
+
+/***************************************************************************/
+/* Known issues:
+ * - FT232H 12MBaud not working
+ *   - using OSX El Capitan 10.11.6, FTDIUSBSerialDriver_v2_3.dmg
+ *   - tried with 2 different chips (FT232H and FT2232H)
+ *   Testing with Ftdi FT232H, which is capable to use up to 12MBaud, only
+ *   line speed up to 3MBaud were accepted. For higher baudrates we earn
+ *   a failure in the ioctl(IOSSIOSPEED) call.
+ *   But as `python -m serial.tools.miniterm` shows the same behaviour, it
+ *   looks that this is a bug or limitation in OSX and/or Ftdi driver.
+ *   Trying with PL2303 (driver version PL2303_MacOSX_1.6.1_20160309.zip),
+ *   baudrates up to 6MBaud were accepted.
+ */
+
+#if defined(__APPLE__) && defined(USE_CUSTOM_BAUD)
+
+#include "custbaud_osx.h"
+#include <string.h>
+#include <stdio.h>
+#include <errno.h>
+#include <sys/ioctl.h>
+#include <IOKit/serial/ioss.h>
+#include "term.h"
+
+/***************************************************************************/
+/* Need to undef tcsetattr to get access to the original tcsetattr()
+ * function inside our module.
+ */
+#undef tcsetattr
```

Comment:
You could also avoid including `custbaud_osx.h`, instead of the undef... Whatever feels better to you.